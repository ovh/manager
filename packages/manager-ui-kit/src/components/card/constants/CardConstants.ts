import { CARD_COLOR, CARD_COLORS } from '@ovhcloud/ods-react';

export { CARD_COLOR, CARD_COLORS };
