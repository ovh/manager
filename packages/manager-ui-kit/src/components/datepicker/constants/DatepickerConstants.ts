import {
  DATEPICKER_DAY,
  DATEPICKER_DAYS,
  DATEPICKER_VIEW,
  DATEPICKER_VIEWS,
} from '@ovhcloud/ods-react';

export { DATEPICKER_DAY, DATEPICKER_DAYS, DATEPICKER_VIEW, DATEPICKER_VIEWS };
