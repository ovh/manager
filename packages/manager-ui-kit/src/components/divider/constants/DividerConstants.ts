import {
  DIVIDER_COLOR,
  DIVIDER_COLORS,
  DIVIDER_SPACING,
  DIVIDER_SPACINGS,
} from '@ovhcloud/ods-react';

export { DIVIDER_COLOR, DIVIDER_COLORS, DIVIDER_SPACING, DIVIDER_SPACINGS };
