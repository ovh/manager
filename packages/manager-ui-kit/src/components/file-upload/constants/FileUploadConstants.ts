import { FILE_REJECTION_CAUSE, FILE_REJECTION_CAUSES, FILE_UPLOAD_I18N } from '@ovhcloud/ods-react';

export { FILE_REJECTION_CAUSE, FILE_REJECTION_CAUSES, FILE_UPLOAD_I18N };
