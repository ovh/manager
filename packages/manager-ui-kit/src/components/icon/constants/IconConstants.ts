import { ICON_NAME, ICON_NAMES } from '@ovhcloud/ods-react';

export { ICON_NAME, ICON_NAMES };
