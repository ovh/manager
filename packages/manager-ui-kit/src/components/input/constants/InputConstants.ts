import {
  INPUT_I18N,
  INPUT_MASK_STATE,
  INPUT_MASK_STATES,
  INPUT_TYPE,
  INPUT_TYPES,
} from '@ovhcloud/ods-react';

export {
  INPUT_I18N,
  INPUT_MASK_STATE,
  INPUT_MASK_STATES,
  INPUT_TYPE,
  INPUT_TYPES,
};
