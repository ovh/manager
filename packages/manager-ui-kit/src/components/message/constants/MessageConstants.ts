import {
  MESSAGE_COLOR,
  MESSAGE_COLORS,
  MESSAGE_I18N,
  MESSAGE_VARIANT,
  MESSAGE_VARIANTS,
} from '@ovhcloud/ods-react';

export { MESSAGE_COLOR, MESSAGE_COLORS, MESSAGE_I18N, MESSAGE_VARIANT, MESSAGE_VARIANTS };
