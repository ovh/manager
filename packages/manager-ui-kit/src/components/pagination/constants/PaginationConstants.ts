import { PAGINATION_PER_PAGE } from '@ovhcloud/ods-react';

export { PAGINATION_PER_PAGE };
