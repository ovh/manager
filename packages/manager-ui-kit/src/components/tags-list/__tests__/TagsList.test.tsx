import React from 'react';

import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { beforeEach, describe, expect, it, vi } from 'vitest';

import { tags } from '@/commons/tests-utils/StaticData.constants';
import { TagsList } from '@/components/tags-list/TagsList.component';

import * as tagsStackUtils from '../tags-stack/TagsStack.utils';

vi.mock('../tags-stack/TagsStack.utils', async (importOriginal) => {
  const actual = await importOriginal();
  const mockedModule: typeof import('../tags-stack/TagsStack.utils') = {
    ...(actual as object),
    getVisibleTagCount: vi.fn(),
  };
  return mockedModule;
});

describe('TagsList Component - Snapshot Tests', () => {
  const modalHeader = 'Test Resource';

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('displays all tags', () => {
    vi.mocked(tagsStackUtils.getVisibleTagCount).mockReturnValue(8);

    const { container } = render(<TagsList tags={tags} modalHeading={modalHeader} />);

    expect(container).toMatchSnapshot();
  });

  it('displays fewer tags along with Link', () => {
    vi.mocked(tagsStackUtils.getVisibleTagCount).mockReturnValue(2);

    const { container } = render(<TagsList tags={tags} maxLines={2} modalHeading={modalHeader} />);

    expect(container).toMatchSnapshot();
  });

  it('displays modal on click of "More tags" link', async () => {
    vi.mocked(tagsStackUtils.getVisibleTagCount).mockReturnValue(1);

    const { baseElement } = render(
      <TagsList tags={tags} maxLines={1} modalHeading={modalHeader} />,
    );
    const moreTagsLink = screen.getByRole('link');
    fireEvent.click(moreTagsLink);
    await waitFor(() => {
      expect(baseElement).toMatchSnapshot();
    });
  });
});
