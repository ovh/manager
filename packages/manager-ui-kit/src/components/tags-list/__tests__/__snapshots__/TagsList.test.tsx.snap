// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TagsList Component - Snapshot Tests > displays all tags 1`] = `
<div>
  <div
    class="w-full h-full flex flex-wrap min-w-[85px]"
  >
    <span
      class="_badge_vpjtq_2 _badge--neutral_vpjtq_27 _badge--md_vpjtq_56 mr-1 mb-1 truncate inline-block"
      data-ods="badge"
      style="max-width: -32px;"
    >
      tag1:tag1
    </span>
    <span
      class="_badge_vpjtq_2 _badge--neutral_vpjtq_27 _badge--md_vpjtq_56 mr-1 mb-1 truncate inline-block"
      data-ods="badge"
      style="max-width: -32px;"
    >
      tag2:tag2
    </span>
    <span
      class="_badge_vpjtq_2 _badge--neutral_vpjtq_27 _badge--md_vpjtq_56 mr-1 mb-1 truncate inline-block"
      data-ods="badge"
      style="max-width: -32px;"
    >
      tag3:tag3
    </span>
    <span
      class="_badge_vpjtq_2 _badge--neutral_vpjtq_27 _badge--md_vpjtq_56 mr-1 mb-1 truncate inline-block"
      data-ods="badge"
      style="max-width: -32px;"
    >
      tag4:tag4
    </span>
  </div>
</div>
`;

exports[`TagsList Component - Snapshot Tests > displays fewer tags along with Link 1`] = `
<div>
  <div
    class="w-full h-full flex flex-wrap min-w-[85px]"
  >
    <span
      class="_badge_vpjtq_2 _badge--neutral_vpjtq_27 _badge--md_vpjtq_56 mr-1 mb-1 truncate inline-block"
      data-ods="badge"
      style="max-width: -32px;"
    >
      tag1:tag1
    </span>
    <span
      class="_badge_vpjtq_2 _badge--neutral_vpjtq_27 _badge--md_vpjtq_56 mr-1 mb-1 truncate inline-block"
      data-ods="badge"
      style="max-width: -32px;"
    >
      tag2:tag2
    </span>
    <span
      class="_badge_vpjtq_2 _badge--neutral_vpjtq_27 _badge--md_vpjtq_56 mr-1 mb-1 truncate inline-block"
      data-ods="badge"
      style="max-width: -32px;"
    >
      tag3:tag3
    </span>
    <a
      class="_link_1y3p4_2 _link--no-text_1y3p4_40 text-xs"
      data-ods="link"
      href="#"
    >
      <span>
        ​
      </span>
      <span
        class="_icon_10c23_2 _icon--chevron-double-right_10c23_82"
        data-ods="icon"
        role="presentation"
      />
    </a>
  </div>
</div>
`;

exports[`TagsList Component - Snapshot Tests > displays modal on click of "More tags" link 1`] = `
<body>
  <div>
    <div
      class="w-full h-full flex flex-wrap min-w-[85px]"
    >
      <span
        class="_badge_vpjtq_2 _badge--neutral_vpjtq_27 _badge--md_vpjtq_56 mr-1 mb-1 truncate inline-block"
        data-ods="badge"
        style="max-width: -32px;"
      >
        tag1:tag1
      </span>
      <span
        class="_badge_vpjtq_2 _badge--neutral_vpjtq_27 _badge--md_vpjtq_56 mr-1 mb-1 truncate inline-block"
        data-ods="badge"
        style="max-width: -32px;"
      >
        tag2:tag2
      </span>
      <a
        class="_link_1y3p4_2 _link--no-text_1y3p4_40 text-xs"
        data-ods="link"
        href="#"
      >
        <span>
          ​
        </span>
        <span
          class="_icon_10c23_2 _icon--chevron-double-right_10c23_82"
          data-ods="icon"
          role="presentation"
        />
      </a>
    </div>
  </div>
</body>
`;
